### DISCUSSION TITLE
Post Office Protocol, Reticulum

### DISCUSSION THREAD

**Discussion by faragher** [markqvist/Reticulum#430] [Show and tell]
This is still very raw and generally undocumented, but this is a mailbox for multiple access to a single LXM destination. It is designed to be run on a trusted system, as it handles unencrypted information locally (all data is encrypted during transmission).

Actual documentation will follow in the coming days:

https://github.com/faragher/POPR

**Comment by faragher** [comment]
I have updated the repository with updated files, removing some unused functions and adding comments, as well as a general specification which likely needs some cleanup and proof-reading.

**Comment by trosel** [comment]
Excellent! So this is a way that we could have "multi-device" usage with a messaging protocol? I guess just have the "PO Box" running on a standalone device or on desktop while you access that via multiple different clients?

**Reply by faragher** [reply]
Correct! I guess I didn't actually specify how it was intended to be used. Bear in mind the server end of this is adequate, but there's no real "client" of any usefulness.

The POPR server is run on a trusted machine, and announces like a normal LXMF client. Any messages sent to it are stored locally. If a POPR client connects, the server checks its identity, and if it's not identified or it is not authorized, it rejects it. Otherwise you can list, download, and delete messages as you please. You can also send a message by feeding it a destination address, content, title, and fields (optionally) and the server will generate and send the message itself. 

As far as the outside world cares, it's just another LXM address, fully compatible with Sideband, NomadNet, telemetry, and anything else that conforms to LXMF specifications.

It's designed for limited multiple access, and will likely bog down with hundreds or thousands of messages, so they should be locally cached and deleted at some point, but this is both a server issue and something POP3 itself faces, so it's up to the user and server to determine what acceptable limits are.

But it's absolutely designed for both multiple-access and local message storage. I'm sure the server/client will need upgrades in the future, but I'm reasonably certain the protocol itself is sound, as I was standing on the shoulders of giants.
